enhancements/oc/mirroring-manifests.md

```diff
@@ -0,0 +1,172 @@
+---
+title: mirroring-manifsts
+authors:
+  - "@wking"
+reviewers:
+  - "@jottofar"
+  - "@LalatenduMohanty"
+  - "@smarterclayton"
+approvers:
+  - "@LalatenduMohanty"
+  - "@smarterclayton"
+creation-date: 2020-01-24
+last-updated: 2020-01-24
+status: provisional|implementable
+---
+
+# Mirroring Manifests
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+For disconnected clusters, `oc adm release mirror ...` already helps users copy the release image and its dependencies into their local mirror.
+This enhancment extends that command to also create and apply manifests to their local cluster, to bring in information that cannot be represented by container images.
+For example, `oc` might produce ConfigMap manifests containing the release image signature which the cluster-version operator could use to [verify the mirrored release][cvo-config-map-signatures].
+
```

Comment:
Do we also want, at least as an option, the ability for 'oc' to do the verification so only verified data is ever sneakernetted onto a possibly highly sensitive network?

Reply:
> ... the ability for 'oc' to do the verification so only verified data is ever sneakernetted onto a possibly highly sensitive network?

Like "does this signature check out?"?  That sounds good to me, but is an internal `oc` decision.  It won't affect the user interface.  Unless you also want flags to turn it on and off?  But I don't know why you'd want flags to turn it off, once `oc` grows the ability to verify signatures.  I guess there is a bit of wiggle room, because `oc`-side verification would be "do the sig stores configured in release B have a valid signature for release B by the keys trusted in release B?" and in a cluster currently running release A, the CVO would be checking "do the sig stores configured in release A have a valid signature for release B by the keys trusted in release A?".  But the point of ConfigMap signatures is that we can populate the ConfigMap sig store configured in release A, so I'm not worried about that.  And if we have concerns about drifting "by the keys trusted in release", we could just include ConfigMaps for all the signatures we can hunt down and let the CVO sort it out inside.

Reply:
+1 for leaving it for CVO for verify the signatures at this point of time. I am still not sure what value `oc` side verfication will add to this.

Reply:
My original question above came out of a lack of understanding of how this ultimately works. However as it turns out in order to "acquire" the signature(s) needed to populate the configmap Verify has to be called. The verification process is what populates a structure containing those signatures that successfully verified the given image. It if doesn't return any signatures (which "should" never happen) something is wrong in which case I just warn the user and proceed with rest of the mirroring task.